cl/opinion_page/views.py

```diff
@@ -97,19 +99,29 @@ def court_homepage(request: HttpRequest, pk: str) -> HttpResponse:
     return TemplateResponse(request, "court.html", render_dict)
 
 
-@group_required("tenn_work_uploaders")
+@group_required(
+    "tenn_work_uploaders", "tennworkcompcl", "tennworkcompapp", "me"
+)
 def court_publish_page(request: HttpRequest, pk: int) -> HttpResponse:
-    """Display upload form and intake Opinions for Tenn. Workers Comp Cl/App
+    """Display upload form and intake Opinions for partner courts
 
     :param request: A GET or POST request for the page
     :param pk: The CL Court ID for each court
     """
+    if pk not in ["tennworkcompcl", "tennworkcompapp", "me"]:
+        raise Http404(
+            "Court pages only implemented for Tennessee Worker Comp Courts and Maine SJC."
+        )
+    # Validate the user has permission
+    if not request.user.is_staff and not request.user.is_superuser:
+        if not [g.name for g in request.user.groups.all() if pk == g.name]:  # type: ignore
```

Comment:
I think you can tweak this to use an `exists()` check instead, which should be faster. Check this out:

https://stackoverflow.com/questions/4789021/in-django-how-do-i-check-if-a-user-is-in-a-certain-group